hca/util/__init__.py

```diff
@@ -507,18 +507,26 @@ def _save_auth_token_refresh_result(self, result):
     def _process_method_args(self, parameters, body_json_schema):
         body_props = {}
         method_args = collections.OrderedDict()
-        for prop_name, prop_data in body_json_schema["properties"].items():
-            enum_values = prop_data.get("enum")
-            type_ = prop_data.get("type") if enum_values is None else 'string'
-            anno = self._type_map[type_]
-            if prop_name not in body_json_schema.get("required", []):
-                anno = typing.Optional[anno]
-            param = Parameter(prop_name, Parameter.POSITIONAL_OR_KEYWORD, default=prop_data.get("default"),
-                              annotation=anno)
-            method_args[prop_name] = dict(param=param, doc=prop_data.get("description"),
-                                          choices=enum_values,
-                                          required=prop_name in body_json_schema.get("required", []))
-            body_props[prop_name] = body_json_schema
+
+        def _parse_properties(properties, schema):
+            for prop_name, prop_data in properties.items():
+                enum_values = prop_data.get("enum")
+                type_ = prop_data.get("type") if enum_values is None else 'string'
+                anno = self._type_map[type_]
+                if prop_name not in body_json_schema.get("required", []):
+                    anno = typing.Optional[anno]
+                param = Parameter(prop_name, Parameter.POSITIONAL_OR_KEYWORD, default=prop_data.get("default"),
+                                  annotation=anno)
+                method_args[prop_name] = dict(param=param, doc=prop_data.get("description"),
+                                              choices=enum_values,
+                                              required=prop_name in body_json_schema.get("required", []))
+                body_props[prop_name] = schema
```

Comment:
I think the main problem might be here. If `allOf` is used, `schema` should be combined with `body_props[prop_name]` instead of replacing it, a la

    body_props[prop_name] = {**schema, **body_props.get(prop_name, {})}

That said, this line won't work as-is, since `schema` and `body_props[prop_name]` need to be recursively / deep merged.